addon/components/connect.js

```diff
@@ -14,13 +14,38 @@ var connect = function(mapStateToComputed, mapDispatchToActions) {
             });
             return props;
         };
+
         var mapDispatch= function(dispatch) {
             var actions = [];
             Object.keys(finalMapDispatchToActions(dispatch)).forEach(function(key) {
                 actions.push(key);
             });
             return actions;
         };
+
+        /**
+         * Transform a key into its private equivalent by prepending an underscore
+         */
+        var getPrivateKey = function(key) {
+            return `_${key}`;
+        };
+
+        /**
+         * Transform properties state object into a private object
+         *
+         * Example:
+         *
+         *     transformPropsToPrivate({ users: [] })  // => { _users: [] }
+         */
+        var transformPropsToPrivate = function(props) {
+            // Transform object to new object with private keys
+            return Object.keys(props).reduce(function(privateState, key) {
+                return Object.assign(privateState, {
```

Comment:
Any advantage to using `Object.assign` here, rather than just assigning `privateState[getPrivateKey(key)] = props[key]`? I just ran a quick test in the console comparing the two and saw a ~50x difference in speed. Granted, I'm not sure how critical a few ms saved here is in the grand scheme of things.